src/classes/ADDR_Addresses_UTIL.cls

```diff
@@ -46,255 +46,298 @@ public with sharing class ADDR_Addresses_UTIL {
     ********************************************************************************************************/
     public static Boolean isContactAddressChanged(Contact con1, Contact con2) {
-        // if both null, no change
-        if (con1 == null && con2 == null) 
-           return false;
+        if (con1 == Null && con2 == Null) {
+            return false;
+        }
 
-        // if one null, make sure other has at least one address field set
-        if (con1 == null || con2 == null) {
-            if (con1 == null) 
-               con1 = con2;
-            return (!isContactAddressEmpty(con1) ||
-               con1.Primary_Address_Type__c != null);
+        if (con1 == Null || con2 == Null) {
+            if (con1 == Null) {
+                con1 = con2;
+            }
+            return (!isContactAddressEmpty(con1) || String.isNotBlank(con1.Primary_Address_Type__c));
         }
         
-        // both provided, so look for at least one change between address fields
-        Boolean isChanged = 
-           !ADDR_Addresses_UTIL.equalsCaseSensitive(con1.MailingStreet, con2.MailingStreet) ||
-           !ADDR_Addresses_UTIL.equalsCaseSensitive(con1.MailingCity, con2.MailingCity) ||
-           !ADDR_Addresses_UTIL.equalsCaseSensitive(con1.MailingState, con2.MailingState) ||
-           !ADDR_Addresses_UTIL.equalsCaseSensitive(con1.MailingPostalCode, con2.MailingPostalCode) ||
-           !ADDR_Addresses_UTIL.equalsCaseSensitive(con1.MailingCountry, con2.MailingCountry) ||
-           !ADDR_Addresses_UTIL.equalsCaseSensitive(con1.Primary_Address_Type__c, con2.Primary_Address_Type__c);
-           
+        Boolean isChanged = !ADDR_Addresses_UTIL.equalsCaseSensitive(con1.MailingStreet, con2.MailingStreet) ||
+                            !ADDR_Addresses_UTIL.equalsCaseSensitive(con1.MailingCity, con2.MailingCity) ||
+                            !ADDR_Addresses_UTIL.equalsCaseSensitive(con1.Mailing_County__c, con2.Mailing_County__c) ||
+                            !ADDR_Addresses_UTIL.equalsCaseSensitive(con1.MailingState, con2.MailingState) ||
+                            !ADDR_Addresses_UTIL.equalsCaseSensitive(con1.MailingPostalCode, con2.MailingPostalCode) ||
+                            !ADDR_Addresses_UTIL.equalsCaseSensitive(con1.MailingCountry, con2.MailingCountry) ||
+                            !ADDR_Addresses_UTIL.equalsCaseSensitive(con1.Primary_Address_Type__c, con2.Primary_Address_Type__c);
+        
         if (!isChanged && ADDR_Addresses_UTIL.isStateCountryPicklistsEnabled) {
-            isChanged = 
-                !ADDR_Addresses_UTIL.equalsCaseSensitive(String.valueOf(con1.get('MailingStateCode')), String.valueOf(con2.get('MailingStateCode'))) || 
-                !ADDR_Addresses_UTIL.equalsCaseSensitive(String.valueOf(con1.get('MailingCountryCode')), String.valueOf(con2.get('MailingCountryCode')));
-        }    
-        //UTIL_Debug.debug('****Contact address changed: ' + isChanged);
```

Comment:
Is this on purpose? Or I should ask if the commenting out was on purpose?

Reply:
@xinnanlisf , apologies but which line are you referring to? Are you referring to why the comment "// both provided, so look for at least one change between address fields" was removed or the UTIL.debug?